src/contexts/TodoContext.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { TodoContextTypes } from '../types/TodoContextTypes';
+import { Status } from '../types/Status';
+import { ErrorsMessage } from '../types/ErrorsMessage';
+import { getFilteredTodos } from '../services/getFilteredTodos';
+import { getTodos } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { USER_ID } from '../constants/User';
+
+export const TodoContext = React.createContext<TodoContextTypes>({
+  todos: [],
+  filteredTodos: [],
+  status: Status.All,
+  tempTodo: null,
+  errorMessage: ErrorsMessage.None,
+  updatingTodosIds: [],
+  addTodo: () => {},
+  deleteTodo: () => {},
+  handleSetTempTodo: () => {},
+  handleSetErrorMessage: () => {},
+  handleUpdatingTodosIds: () => {},
+  changeStatus: () => {},
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodoProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<ErrorsMessage>(
+    ErrorsMessage.None,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [updatingTodosIds, setUpdatingTodosIds] = useState<number[]>([]);
+
+  const filteredTodos = getFilteredTodos(todos, status);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorsMessage.Load);
+      });
+  }, []);
```

Comment:
Consider adding some loader logic to show loader until todo aren`t set

Reply:
I didn't really understand this point. What do I need to do?

Reply:
It's better to ask questions in the chat because we don't have notifications here. It's okay, I was just suggesting making some kind of downloader